routes/user.js

```diff
@@ -268,53 +247,56 @@ router.route('/reset/:token')
+        req.checkBody('confirm', 'Passwords do not match').equals(req.body.password);
+        var errors = req.validationErrors()
+        if (errors) {
+            res.render('reset', {
+                errors: errors
+            })
+        } else {
+            async.waterfall([
+                function(done) {
+                    User.findOne({
+                        resetPasswordToken: req.params.token,
+                        resetPasswordExpires: {
+                            $gt: Date.now()
+                        }
+                    }, function(err, user) {
+                        if (!user) {
+                            req.flash('error', 'Password reset token is invalid or has expired.');
+                            return res.redirect('back');
+                        }
 
-                    user.password = user.generateHash(req.body.password);
-                    user.resetPasswordToken = undefined;
-                    user.resetPasswordExpires = undefined;
+                        user.password = user.generateHash(req.body.password);
+                        user.resetPasswordToken = undefined;
+                        user.resetPasswordExpires = undefined;
 
-                    user.save(function(err) {
-                        req.logIn(user, function(err) {
-                            done(err, user);
+                        user.save(function(err) {
+                            req.logIn(user, function(err) {
+                                done(err, user);
+                            });
                         });
                     });
-                });
-            },
-            function(user, done) {
-                var smtpTransport = nodemailer.createTransport('SMTP', {
-                    service: 'SendGrid',
-                    auth: {
-                        user: process.env.SENDGRID_USER || config.get('emailVerification.user'),
-                        pass: process.env.SENDGRID_PASS || config.get('emailVerification.pass')
-                    }
-                });
-                var mailOptions = {
-                    to: user.email,
-                    from: 'noreply@theotech.org',
-                    subject: 'Your password has been changed',
-                    text: 'Hello,\n\n' +
-                        'This is a confirmation that the password for your account ' + user.email + ' has just been changed.\n'
-                };
-                smtpTransport.sendMail(mailOptions, function(err) {
-                    req.flash('success', 'Success! Your password has been changed.');
-                    done(err);
-                });
-            }
-        ], function(err) {
-            res.redirect('/');
-        });
+                },
+                function(user, done) {
+                    var mailOptions = {
+                        to: user.email,
+                        from: 'noreply@theotech.org',
```

Comment:
assuming you're going to change the from address? hopefully i won't be hard to merge with the other code change you already made.